pkg/cloud/vsphere/services/userdata/controlplane.go

```diff
@@ -24,6 +24,28 @@ import (
 )
 
 const (
+	cloudConfig = `[Global]
+insecure-flag = "1" # set to 1 if the vCenter uses a self-signed cert
+datacenters = "{{ .Datacenter }}"
+
+[VirtualCenter "{{ .Server }}"]
+user = "{{ .User }}"
```

Comment:
Hi @andrewsykim,

I really don't want to send the user and pass via cloud-init in user data. Please set the Kubernetes secret name for the cloud provider credentials and then add that secret to the target cluster.

Reply:
This [doc](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/existing.html#supported-parameters) discusses the params for pointing to the secret and the expected format, etc.